docusaurus/docs/React/guides/sdk-state-management.mdx

```diff
@@ -0,0 +1,119 @@
+---
+id: sdk-state-management
+title: SDK State Management
+---
+
+Most of the application state which is the main driver behind the UI changes of our chat SDK lives within our LLC ([`StreamChat`](https://github.com/GetStream/stream-chat-js)) - more specifically client and channel instances and is _NOT REACTIVE_ - our React SDK makes these states reactive by listening to the same events client does and copies it with the help of `useState` API after it has been changed.
+
+This is how the SDK state pipeline looks like behind the scenes:
+
+```txt
+client attaches WS event listeners -> React SDK attaches WS event listeners -> WS event received -> client/channel instance updates its own state -> React SDK pulls the updated state and updates it locally
+```
+
+Optimistic updates - if applicable - are handled by the React SDK.
+
+TODO: link to optimistic update tutorial?
```

Comment:
Not sure, if we should put optimistic state updates tutorial. Not sure whether we have some standardized approach to the topic, that could be documented.